apps/swirl-docs/src/components/Layout/DocumentationLayout.tsx

```diff
@@ -1,62 +1,61 @@
-import { FrontMatter } from "@swirl/lib/docs/src/docs.model";
-import { NavItem } from "@swirl/lib/navigation";
-import { MDXRemote } from "next-mdx-remote";
 import { CategoryNav } from "./CategoryNav";
 import { DocLinksNav } from "./DocLinksNav";
 import Footer from "./Footer";
 import { DocumentationHeader } from "../Documentation/DocumentationHeader";
 import { ComponentPreview } from "../ComponentPreview";
 import { useToC } from "@swirl/lib/hooks/useToC";
 import classNames from "classnames";
-
-export type ComponentExample = {
-  description: string;
-  url: string;
-  title: string;
-};
+import DocumentationLayoutContext, {
+  TDocumentationLayout,
+} from "./DocumentationLayoutContext";
+import { ReactNode } from "react";
+import MDXDocument from "./MDXDocument";
 
 interface DocumentationLayoutProps {
-  categoryLinkList: NavItem[] | undefined;
-  document: any;
-  mdxComponents?: any;
-  frontMatter?: FrontMatter;
+  data: TDocumentationLayout;
+  content: ReactNode | ReactNode[];
+  header?: ReactNode | ReactNode[];
+  footer?: ReactNode | ReactNode[];
 }
 
-export const DocumentationLayout = ({
-  categoryLinkList,
-  mdxComponents,
-  document,
-  frontMatter,
-}: DocumentationLayoutProps) => {
-  const hasFrontMatterTitle = frontMatter?.title;
-  const isComponentDoc = frontMatter?.examples ? true : false;
-
-  const [tocItems] = useToC(document, isComponentDoc);
+export function DocumentationLayout({
+  header,
+  content,
+  footer,
+  data,
+}: DocumentationLayoutProps) {
+  const [tocItems] = useToC(data.mdxContent?.document!, false);
 
   return (
-    <div className="flex">
-      <CategoryNav categoryLinkList={categoryLinkList} />
-      <div className="h-full w-full">
-        <main
-          id="main"
-          className={classNames(
-            "grid grid-cols-1 md:grid-cols-[minmax(0,_45rem)_16rem] gap-8 justify-center",
-            "my-0 mx-auto mt-14 mb-4 md:mb-0 px-4 lg:px-0"
-          )}
-        >
-          <article className="w-full max-w-[45rem]">
-            {hasFrontMatterTitle && (
-              <DocumentationHeader frontMatter={frontMatter} />
+    <DocumentationLayoutContext.Provider value={data}>
+      <div className="flex">
+        {data.navigationLinks && <CategoryNav />}
+        <div className="h-full w-full">
+          <main
+            id="main"
+            className={classNames(
+              "grid grid-cols-1 md:grid-cols-[minmax(0,_45rem)_16rem] gap-8 justify-center",
+              "my-0 mx-auto mt-14 mb-4 md:mb-0 px-4 lg:px-0"
+            )}
+          >
+            <article className="w-full max-w-[45rem]">
```

Comment:
**question:** Is it safe to assume that this can always be an article? What if you want to have multiple articles inside. `content`?